hazelcast-sql/src/main/java/com/hazelcast/sql/impl/type/QueryDataType.java

```diff
@@ -272,56 +255,47 @@ private void collectNestedTypes(final QueryDataType dataType, final Map<String,
 
     @Override
     public void readData(ObjectDataInput in) throws IOException {
-        this.converter = Converters.getConverter(in.readInt());
-        if (!converter.getTypeFamily().equals(QueryDataTypeFamily.OBJECT)) {
+        converter = Converters.getConverter(in.readInt());
+        if (converter.getTypeFamily() != QueryDataTypeFamily.OBJECT) {
             return;
         }
 
-        this.objectTypeKind = in.readInt();
-        this.objectTypeName = in.readString();
-        this.objectTypeMetadata = in.readString();
+        objectTypeKind = TypeKind.of(in.readInt());
+        objectTypeName = in.readString();
+        objectTypeMetadata = in.readString();
+
         if (!isCustomType()) {
             return;
         }
 
-        this.objectFields = new ArrayList<>();
-        final int typeMapSize = in.readInt();
-        final Map<String, QueryDataType> nestedTypes = new HashMap<>();
+        objectFields = new ArrayList<>();
+        int typeMapSize = in.readInt();
+        Map<String, QueryDataType> nestedTypes = new HashMap<>();
         nestedTypes.put(objectTypeName, this);
 
         for (int i = 0; i < typeMapSize; i++) {
-            final Integer currentTypeKind = in.readInt();
-            final String currentTypeName = in.readString();
-            final String currentTypeMetadata = in.readString();
-
-            final int fieldsSize = in.readInt();
-            final QueryDataType currentType = nestedTypes.computeIfAbsent(currentTypeName, s -> {
-                // TODO: simplify?
-                final QueryDataType fieldType = new QueryDataType(currentTypeName);
-                fieldType.setObjectTypeMetadata(currentTypeMetadata);
-                fieldType.setObjectTypeKind(currentTypeKind);
-                return fieldType;
-            });
+            TypeKind currentTypeKind = TypeKind.of(in.readInt());
```

Comment:
I do not think that removing `final` improves readability, on the contrary.

Reply:
I have refactored `readData()` to make it symmetric to `writeData()`. The final version is halved in length and quite readable without `final`.